compiler/fir/analysis-tests/testData/enhancement/jsr305/Strange.fir.txt

```diff
@@ -1,9 +1,9 @@
 public open class Strange : R|kotlin/Any| {
-    @R|javax/annotation/Nonnull|(R|javax/annotation/meta/When.UNKNOWN|()) public open field field: R|ft<kotlin/String, kotlin/String?>!|
+    @R|javax/annotation/Nonnull|(R|javax/annotation/meta/When.UNKNOWN|()) public open field field: R|ft<@FlexibleNullability kotlin/String, kotlin/String?>!|
```

Comment:
Most of analysis test data changes are about adding either `@FlexibleNullability` or `@EnhancedNullability`. As you can see below, `@FlexibleNullability` is added only if no attributes are found, and then just recognized a few case where `@EnhancedNullability` should be attached. I just extended the logic to load attributes from annotations to cover more non-null annotations.

Anyway, my high-level question is, should this attribute be bound to upper type as well? Currently, it's only associated with lower type.

Reply:
> my high-level question is, should this attribute be bound to upper type as well? Currently, it's only associated with lower type.

Answer to my own question: since this attribute is added to flexible type whose lower/upper bounds' nullability don't match, it makes sense to add this to upper bound instead. However, as long as the retrieval of attributes takes into account both lower and upper bound, it doesn't matter technically. Such `union` logic is just added in the 2nd commit.

Reply:
Correction to my own answer: this should be recorded at lower bound, since the attribute of flexible type is supposed to be that of lower bound.